### PR TITLE
Add Edit this page button to docs

### PR DESCRIPTION
**Pull request by snowystinger**
This should make it easier for people to contribute doc fixes
I pass the filepath through and store it in a meta tag in a data attribute, so it shouldn't conflict with anything now or in the future
We could limit the pages that it appears on, but I haven't done that right now

Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Review by ktabors — DISMISSED** [review]
lgtm

**Comment by ross-pfahler** [comment]
hey -- do we want to get this in? seems like a good change.

my only question -- what do we think about moving this to the bottom right? Like more inline with the TOC - but at the bottom. Feels like it will be less intrusive.

**Review by LFDanLu — APPROVED** [review]
LGTM, only opinion would that moving the edit button to the top of the page might make it more obvious for users, but I've seen various docs sites have it at the bottom/top so not too chuffed about what we go with here.

**Review by ktabors — COMMENTED** [review]
lgtm

**Review by ktabors — APPROVED** [review]

**Comment by devongovett** [comment]
But... why?

File: packages/dev/docs/src/client.js [hunk 1]
```diff
@@ -182,6 +184,12 @@ ReactDOM.render(<>
   <ThemeSwitcher />
 </>, document.querySelector('.' + docsStyle.pageHeader));
 
+let pathToPage = document.querySelector('[data-github-src]').getAttribute('data-github-src');
+ReactDOM.render(<Button variant="primary" isQuiet elementType="a" href={encodeURI(`https://github.com/adobe/react-spectrum/tree/main/${encodeURI(pathToPage)}`)} target="_blank">
+  <LinkOut />
+  <Text>Edit this page</Text>
+</Button>, document.querySelector('#edit-page'));
```
Comment (devongovett) [inline packages/dev/docs/src/client.js:191]: Why does this need to be client rendered? Why not an `a` tag with button css on it?

Comment (snowystinger) [reply packages/dev/docs/src/client.js:191]: honestly, it's just how old this PR is I think, we've talked a lot about links vs buttons in the last year, I'll go ahead and update
as for "why", i think it makes it a bit more inviting and we've had a number of issues (not huge, but also not zero) where people point out issues in our docs, this just makes it a little easier for people to actually make the change.

**Review by majornista — APPROVED** [review]

**Review by ktabors — DISMISSED** [review]
Still looks good. Do we want the url to end with`?plain=1` so it's "raw" mode which feels more like an edit mode?

**Comment by snowystinger** [comment]
> Still looks good. Do we want the url to end with`?plain=1` so it's "raw" mode which feels more like an edit mode?

Going to just follow the parcel example for now.

**Comment by LFDanLu** [comment]
couple of conflicts on this PR, but looks fine to me in its current state. Did you also want to go ahead and change the button to an `<a>` with button css?

**Review by yihuiliao — APPROVED** [review]
seems to work well! tested it on a couple pages

**Review by reidbarber — APPROVED** [review]

**Pull request merged**